Source/WebCore/loader/MixedContentChecker.cpp

```diff
@@ -206,7 +206,7 @@ bool MixedContentChecker::canModifyRequest(const URL& url, FetchOptions::Destina
     if (url.protocolIs("https"_s))
         return false;
     // 4.1.2 request’s URL’s host is an IP address.
-    if (!shouldUpgradeIPAddressAndLocalhostForTesting && URL::hostIsIPAddress(url.host()))
+    if (!shouldUpgradeIPAddressAndLocalhostForTesting && !shouldTreatAsPotentiallyTrustworthy(url) && URL::hostIsIPAddress(url.host()))
```

Comment:
Doesn't this make `shouldUpgradeIPAddressAndLocalhostForTesting` kind of meaningless? Since whether it's true or false, `shouldTreatAsPotentiallyTrustworthy` will return true for localhost and IP addresses.

Reply:
Cool. Thanks. And I think this order is much easier to read.
```
    // 4.1.2 request’s URL’s host is an IP address.
    if (URL::hostIsIPAddress(url.host()) && !shouldTreatAsPotentiallyTrustworthy(url))
```